### PR TITLE
[#167347769] update approach to modal for ContextualHelp

### PR DESCRIPTION
**Pull request by valentinacala**
This pr updates some screens and components so that the Contextual Helps are rendered using the light modal approach instead of the modal component.

Affected items rendering a contextual help:
- PaymentMethodsList 
- TransactionDetailsScreen
- ConfirmPaymentMethodScreen
- BaseScreenComponent

The same pr removes the contextual help showing the ToS from SpidInformationScreen

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#167347769](https://www.pivotaltracker.com/story/show/167347769): L'unico item ad utilizzare il componente <Modal> deve essere IdentificationModal

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by valentinacala** [comment]
> It doesn't seem to work in identification.
> The contextual help body is under the identification modal.
> Could the modal be rendered over all other views and causing this problem?
> 

@Undermaken the pr is now updated so that the BaseScreenComponent continues to use a modal approach to the contextual help rendering.

**Comment by Undermaken** [comment]
I tried the following solution and it seems to work (IdentificationModal + ContextHelp)
Could you check if it could have some issues?

#### BaseScreenComponent.tsx
```typescript
import { Container } from "native-base";
import { connectStyle } from "native-base-shoutem-theme";
import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
import * as React from "react";

import { ContextualHelp } from "../ContextualHelp";
import { SearchType } from "../search/SearchButton";
import { BaseHeader } from "./BaseHeader";

interface ContextualHelpProps {
  title: string;
  body: () => React.ReactNode;
}

interface OwnProps {
  contextualHelp?: ContextualHelpProps;
  headerBody?: React.ReactNode;
  appLogo?: boolean;
  isSearchAvailable?: boolean;
  searchType?: SearchType;
}

type BaseHeaderProps =
  | "dark"
  | "appLogo"
  | "primary"
  | "goBack"
  | "headerTitle"
  | "onShowHelp"
  | "body"
  | "isSearchAvailable"
  | "searchType";

type Props = OwnProps &
  Pick<React.ComponentProps<typeof BaseHeader>, BaseHeaderProps>;

interface State {
  isHelpVisible: boolean;
}

class BaseScreenComponent extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      isHelpVisible: false
    };
  }

  private showHelp = () => {
    this.setState({ isHelpVisible: true });
  };

  private hideHelp = () => {
    this.setState({ isHelpVisible: false });
  };

  public render() {
    const {
      dark,
      appLogo,
      contextualHelp,
      goBack,
      headerBody,
      headerTitle,
      primary,
      isSearchAvailable,
      searchType
    } = this.props;
    if (contextualHelp && this.state.isHelpVisible) {
      return (
        <ContextualHelp
          title={contextualHelp.title}
          body={contextualHelp.body}
          onClose={this.hideHelp}
        />
      );
    }
    return (
      <Container>
        <BaseHeader
          primary={primary}
          dark={dark}
          goBack={goBack}
          headerTitle={headerTitle}
          onShowHelp={contextualHelp ? this.showHelp : undefined}
          isSearchAvailable={isSearchAvailable}
          searchType={searchType}
          body={headerBody}
          appLogo={appLogo}
        />
        {this.props.children}
      </Container>
    );
  }
}

export default connectStyle(
  "UIComponent.BaseScreenComponent",
  {},
  mapPropsToStyleNames
)(BaseScreenComponent);
```

**Comment by valentinacala** [comment]
@Undermaken I tested the proposed approach and included it within e78674c634d030a38b31d3c44af38f854b13cade

**Comment by Undermaken** [comment]
@matteodesanti 
Since the new approach to show contextual help consists of a view rendered over the host one, it can't fit the entire screen and for most cases the bottom bar remains always displayed.
Is it ok for you?

**Comment by Undermaken** [comment]
@valentinacala I'm reviewing this [PR](https://github.com/teamdigitale/io-app/pull/1270/files) while I noticed some screens are using **_LightModalContext_**. So I made some tests and, about [this](https://github.com/teamdigitale/io-app/pull/1229#issuecomment-522626207) problem, I produced the following solution (indeed using lightModalContext) 
What do you think?
### BaseScreenComponent.tsx
```typescript
import { Container } from "native-base";
import { connectStyle } from "native-base-shoutem-theme";
import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
import * as React from "react";

import { ContextualHelp } from "../ContextualHelp";
import { SearchType } from "../search/SearchButton";
import { BaseHeader } from "./BaseHeader";
import { withLightModalContext } from "../helpers/withLightModalContext";
import { LightModalContextInterface } from "../ui/LightModal";

interface ContextualHelpProps {
  title: string;
  body: () => React.ReactNode;
}

interface OwnProps {
  contextualHelp?: ContextualHelpProps;
  headerBody?: React.ReactNode;
  appLogo?: boolean;
  isSearchAvailable?: boolean;
  searchType?: SearchType;
}

type BaseHeaderProps =
  | "dark"
  | "appLogo"
  | "primary"
  | "goBack"
  | "headerTitle"
  | "onShowHelp"
  | "body"
  | "isSearchAvailable"
  | "searchType";

type Props = OwnProps &
  Pick<React.ComponentProps<typeof BaseHeader>, BaseHeaderProps> &
  LightModalContextInterface;

interface State {
  isHelpVisible: boolean;
}

class BaseScreenComponent extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      isHelpVisible: false
    };
  }

  private showHelp = () => {
    const { contextualHelp } = this.props;
    if (!contextualHelp) {
      return null;
    }
    this.props.showModal(
      <ContextualHelp
        title={contextualHelp.title}
        body={contextualHelp.body}
        onClose={this.hideHelp}
      />
    );
    //this.setState({ isHelpVisible: true });
  };

  private hideHelp = () => {
    //this.setState({ isHelpVisible: false });
    this.props.hideModal();
  };

  public render() {
    const {
      dark,
      appLogo,
      contextualHelp,
      goBack,
      headerBody,
      headerTitle,
      primary,
      isSearchAvailable,
      searchType
    } = this.props;
    if (contextualHelp && this.state.isHelpVisible) {
      return (
        <ContextualHelp
          title={contextualHelp.title}
          body={contextualHelp.body}
          onClose={this.hideHelp}
        />
      );
    }
    return (
      <Container>
        <BaseHeader
          primary={primary}
          dark={dark}
          goBack={goBack}
          headerTitle={headerTitle}
          onShowHelp={contextualHelp ? this.showHelp : undefined}
          isSearchAvailable={isSearchAvailable}
          searchType={searchType}
          body={headerBody}
          appLogo={appLogo}
        />
        {this.props.children}
      </Container>
    );
  }
}

export default connectStyle(
  "UIComponent.BaseScreenComponent",
  {},
  mapPropsToStyleNames
)(withLightModalContext(BaseScreenComponent));
```

**Comment by valentinacala** [comment]
@Undermaken the last approach you mentioned works well with app screens with the exception of the pin screen:  it is rendered as modal and, applying the light modal approach, its contextual help is not rendered as first element:

helppbug

**Comment by Undermaken** [comment]
> @Undermaken the last approach you mentioned works well with app screens with the exception of the pin screen: it is rendered as modal and, applying the light modal approach, its contextual help is not rendered as first element:
> 
> helppbug

I always forget that case :sweat:
I felt I was missing something, thanks for the explanation

**Comment by valentinacala** [comment]
https://github.com/teamdigitale/io-app/pull/1229/commits/1024f8c065181b07d0a0721d5b1297431574a17f makes the BaseScreenComponent uses again a modal to render the contextual help

File: ts/screens/wallet/TransactionDetailsScreen.tsx [hunk 1]
```diff
@@ -211,6 +209,17 @@ class TransactionDetailsScreen extends React.Component<Props> {
     }
   };
 
+  private showHelp = () => {
+    // tslint:disable-next-line:no-unused-expression
```
[the remark this answers is not in the document]
Comment (valentinacala) [reply ts/screens/wallet/TransactionDetailsScreen.tsx]: removed (https://github.com/teamdigitale/io-app/pull/1229/commits/045087f7941e85dbddfdd23cf55563c8c407337b)

**Pull request merged**